synthea-hiv/uploader/fhir_hook_test.py

```diff
@@ -0,0 +1,59 @@
+import unittest
+from unittest import mock
+import fhir_hook
+import log_mixin
+
+
+def mock_requests_post(**kwargs):
+
+  class MockResponse:
+
+    def __init__(self, text, status_code):
+      self.text = text
+      self.status_code = status_code
+      
+      if status_code >= 400:
+        self.request = mock.MagicMock()
+        self.request.body  = "this happened"
+
+  if 'localhost' in kwargs['url']:
+    return MockResponse('{"resourceType":"Local"}', 200)
+  elif 'google' in kwargs['url']:
+    return MockResponse('{"resourceType":"Google"}', 201)
+
+  return MockResponse(None, 404)
+
+
+class FhirHookTest(unittest.TestCase):
+
+  def setUp(self):
+    super().setUp()
+    mock.patch(
+        'google.auth.transport.requests.Request',
+        return_value=mock.MagicMock()).start()
+
+    mock.patch(
+        'google.auth.default', return_value=(mock.MagicMock(), '_')).start()
+
+    mock.patch('requests.post', side_effect=mock_requests_post).start()
+
+  def test_post_data_local(self):
+    hook = fhir_hook.OpenMRSHook('http://localhost')
+    hook.post_data(resource='Patient', data={'hi': 'mom'})
+    self.assertEqual(hook.response, {'resourceType': 'Local'})
+
+  def test_post_data_gcp(self):
+    hook = fhir_hook.GcpHook('http://googleapis')
+    hook.post_data(data={'hi': 'mom'})
```

Comment:
I am removing `**kwargs`, but if I didn't I had the line:

`kwargs.pop(resrource,''). `

so if resource did not exist, it would return an empty string